pkg/frontend/admin_openshiftcluster_kubernetespods_logs_test.go

```diff
@@ -0,0 +1,212 @@
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+		{
+			name:           "empty container name",
+			namespace:      "openshift-namespace",
+			podName:        "config",
+			containerName:  "",
+			mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},
+			wantStatusCode: http.StatusBadRequest,
+			wantError:      "400: InvalidParameter: : The provided container name '' is invalid.",
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+	} {
+		t.Run(fmt.Sprintf("%s: %s", method, tt.name), func(t *testing.T) {
+			resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+			ti := newTestInfra(t).WithOpenShiftClusters().WithSubscriptions()
+			defer ti.done()
+			k := mock_adminactions.NewMockKubeActions(ti.controller)
+
+			tt.mocks(tt, k)
+
+			ti.fixture.AddOpenShiftClusterDocuments(&api.OpenShiftClusterDocument{
+				Key: strings.ToLower(resourceID),
+				OpenShiftCluster: &api.OpenShiftCluster{
+					ID:   resourceID,
+					Name: "resourceName",
+					Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+					Properties: api.OpenShiftClusterProperties{
+						NetworkProfile: api.NetworkProfile{
+							APIServerPrivateEndpointIP: "0.0.0.0",
+						},
+					},
+				},
+			})
+			ti.fixture.AddSubscriptionDocuments(&api.SubscriptionDocument{
+				ID: mockSubID,
+				Subscription: &api.Subscription{
+					State: api.SubscriptionStateRegistered,
+					Properties: &api.SubscriptionProperties{
+						TenantID: mockTenantID,
+					},
+				},
+			})
+
+			err := ti.buildFixtures(nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, api.APIs, &noop.Noop{}, nil, func(*logrus.Entry, env.Interface, *api.OpenShiftCluster) (adminactions.KubeActions, error) {
+				return k, nil
+			}, nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			go f.Run(ctx, nil, nil)
+
+			resp, b, err := ti.request(method,
+				fmt.Sprintf("https://server/admin%s/kubernetespodlogs?container=%s&namespace=%s&name=%s", resourceID, tt.containerName, tt.namespace, tt.podName),
+				nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = validateResponse(resp, b, tt.wantStatusCode, tt.wantError, tt.wantResponse)
+			if err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestValidateAdminKubernetesPodLogs(t *testing.T) {
+	longName := strings.Repeat("x", 256)
+
+	for _, tt := range []struct {
+		test          string
+		containerName string
+		namespace     string
+		name          string
+		wantErr       string
+	}{
+		{
+			test:          "valid openshift namespace",
+			namespace:     "openshift",
+			containerName: "container-01",
+			name:          "Valid-NAME-01",
+		},
+		{
+			test:          "invalid customer namespace",
```

Comment:
nit: Namespace is valid, but we do not allow access into it. So I would remove `invalid` from the test case name.


```suggestion
			test:          "customer namespace",
```